### PR TITLE
Change include layout

### LINKED ISSUE #2: Change source layout in include/
The `include/` directory does currently not have a `pathie/` subdirectory in the repository itself; it only appears when running `make install` and/or building the release tarball. This makes use of Pathie as Git submodule complicated because it breaks include pathes.

The problem should be fixed by adding a `pathie/` subdirectory under `include/` right in the source tree.

### PR DESCRIPTION
**Pull request by TryExceptElse**
Move declaration files from include/ to include/pathie/.

Closes #2

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -152,17 +152,17 @@ if (CMAKE_BUILD_SHARED_LIBS)
 endif()
 
 install(FILES
-  "include/errors.hpp"
-  "include/entry_iterator.hpp"
-  "include/path.hpp"
-  "include/temp.hpp"
-  "include/pathie.hpp"
+  "include/pathie/errors.hpp"
+  "include/pathie/entry_iter`ator.hpp"
```
Comment (Quintus) [inline CMakeLists.txt]: The backtrick probably shouldn't be there.

**Comment by Quintus** [comment]
Thanks for your contribution. The deprecated headers are a nice idea. I wouldn't have created them as they are only relevant for use of Pathie directly from the build directory (which is rather unofficial use), but they certainly don't hurt.

However, is there a specific reason why you use `#pragma message` instead of `#warning` in the deprecation headers?

Also note the backtick problem I pointed out in a previous comment in the diff.

I'll try running the tests later on.

File: CMakeLists.txt (hunk 1 above)
Comment (TryExceptElse) [reply CMakeLists.txt]: It definitely should not be. It is removed by e6917b8.

**Comment by TryExceptElse** [comment]
> However, is there a specific reason why you use `#pragma message` instead of `#warning` in the deprecation headers?

`#warning` is not supported by all compilers (notably MSVC), and will cause an error when they encounter it. It could be enclosed in an `#ifdef` which determines the compiler, but this seemed sufficient.

**Pull request merged**